pkg/gatewayapi/http_route.go

```diff
@@ -0,0 +1,51 @@
+package gatewayapi
+
+import (
+	"github.com/getkin/kin-openapi/openapi3"
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func HTTPRouteMatchesFromOAS(doc *openapi3.T) ([]gatewayapiv1alpha2.HTTPRouteMatch, error) {
+	httpRouteMatches := []gatewayapiv1alpha2.HTTPRouteMatch{}
+	pathMatchPathPrefix := gatewayapiv1alpha2.PathMatchPathPrefix
+	headerMatchType := gatewayapiv1alpha2.HeaderMatchExact
+	queryParamMatchExact := gatewayapiv1alpha2.QueryParamMatchExact
+
+	for path, pathItem := range doc.Paths {
+		for verb, operation := range pathItem.Operations() {
+			pathValue := path
+			path := &gatewayapiv1alpha2.HTTPPathMatch{
+				Type:  &pathMatchPathPrefix,
+				Value: &pathValue,
+			}
+
+			headers := []gatewayapiv1alpha2.HTTPHeaderMatch{}
+			queryParams := []gatewayapiv1alpha2.HTTPQueryParamMatch{}
+
+			for _, parameter := range operation.Parameters {
```

Comment:
Should the matchers include request data beyond HTTP method and path?

In other words, as a protecting layer, if the spec says the request includes some header, should the request be rejected if the header is not there?

Reply:
That is an option from the Gateway API specification, if the request doesn't match the criteria it will be forwarded to a matcher that only specifies the path and doesn't include headers 

https://gateway-api.sigs.k8s.io/v1alpha2/guides/http-routing/

Reply:
After looking at the OAS, there are some parameters that are required, so it make sense to protect the API in that sense and reject requests without required parameters. 

In your implementation there are two issues:
a) check the parameter is required or not (only add a matcher if the `required` is `true`
b) Pathitems also have parameters that are inherited by all operations for that path.

Reply:
Added a check for required and to loop through the `pathItem` inherited parameters